DataFormats/Common/interface/RefToBase.h

```diff
@@ -250,9 +250,11 @@ namespace edm {
   RefToBase<T>::key() const
   {
     if ( holder_ == nullptr )
-	Exception::throwThis(errors::InvalidReference,
-	  "attempting get key from  null RefToBase;\n"
-	  "You should check for nullity before calling key().");
+    {
+        Exception::throwThis(errors::InvalidReference,
+          "attempting get key from  null RefToBase;\n"
+          "You should check for nullity before calling key().");
+    }
```

Comment:
Random insignificant change here. It's just that static analyzer is not capable of grasping what is in the `if` statement without the braces. It thinks `return  holder_->key();` is also executed in the `if` branch, thus raising about 10 "bugs" which are actually none.

Example:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/ib-static-analysis/CMSSW_10_3_X_2018-09-26-1100/slc6_amd64_gcc700/llvm-analysis/report-daae7d.html#EndPath

Reply:
No, I think that the exception will be resolved if you return a null pointer inside the braces (for example), and as such you do not execute the next call which evaluates```holder_->key()```.

As it is now, if ```holder_``` is a ```nullptr```, then ```holder_->key()``` will fail, also after your "fix"

Reply:
Ok, in other words, that means the analyzer just things execution goes on after throwing an exception? So then it's not completely trivial, let me just revert here. PR got too big anyway.

Reply:
I suspect the static analyzer doesn't necessarily know that `edm::Exception::throwThis()` throws an exception.

Reply:
Right, that could be. But the actual problem can be just solved with an `if` + `else`, no?

Reply:
```
    if ( holder_ == nullptr )
    {
	Exception::throwThis(errors::InvalidReference,
	  "attempting get key from  null RefToBase;\n"
	  "You should check for nullity before calling key().");
    } else {
        return holder_->key();
    }
```

Reply:
As such it will not return, and I don't know if you are going to ask for more troubles from the analyzer...
Why not to simply add a dummy ``` return nullptr``` inside the braces, and make the static analyzer happy?

Reply:
Doesn't it have to return a `size_t`? Can you do `return nullptr` then? Or maybe just return -1?

Reply:
Indeed... You understood what I meant, anyhow ;-)

Reply:
Yes, sorry for being the smartass I am :)